PYIC-8046: Clean up old lambda code + processCandidateIdentity feature flag

AmritSidhu opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Now that the process-candidate-identitylambda has been turned on, we should clean up the old lambdas that are no longer being used

Remove processCandidateIdentity featureFlag from journeyMap
Remove call-ticf-cri lambda
Remove check-coi lambda
Remove store-identity lambda
Remove evaluate-gpg45-scores lambda
Remove additional calls/references to removed 

Subsequent PR: https://github.com/govuk-one-login/ipv-core-common-infra/pull/1191

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-8046](https://govukverify.atlassian.net/browse/PYIC-8046)

## Checklists

### Environment variables or secrets

<!-- Delete if changes DO include new environment variables or secrets -->
- [x] No environment variables or secrets were added or changed


[PYIC-8046]: https://govukverify.atlassian.net/browse/PYIC-8046?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

AmritSidhu committed:
PYIC-8046: remove processCandidateIdentity featureFlag from journeyMap

processCandidateIdentity is now live

AmritSidhu committed:
PYIC-8046: Remove call-ticf-cri lambda

AmritSidhu committed:
PYIC-8046: Remove check-coi lambda

AmritSidhu committed:
PYIC-8046: remove store-identity lambda

AmritSidhu committed:
PYIC-8046: remove evaluate-gpg45-scores lambda

AmritSidhu committed:
PYIC-8046: Remove additional calls/references to removed lambdas

blakeyp commented on lambdas/check-existing-identity/src/test/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandlerTest.java:
```diff
@@ -327,51 +322,6 @@ void shouldUseEvcsService() throws Exception {
 
-        @ParameterizedTest
-        @EnumSource(names = {"M1A", "M1B", "M2B"})
-        void shouldReturnJourneyReuseResponseIfScoresSatisfyGpg45Profile(
```
Isn't this test still relevant, just need to remove the `IPV_GPG45_PROFILE_MATCHED` audit assertions?

blakeyp commented on lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/failed.yaml:
```diff
@@ -564,6 +340,11 @@ states:
       next:
         targetState: RETURN_TO_RP
 
+  FAILED_NO_TICF:
```
Is this used anywhere?

AmritSidhu replied:
This is used a lot across all the journeyMaps. It does seem like a redundant step as its a stepping stone to go directly to: 
```
NO_MATCH_PAGE:
    response:
      type: page
      pageId: pyi-no-match
    events:
      next:
        targetState: RETURN_TO_RP
```

I could refactor to call the NO_MATCH_PAGE directly or keep as is for now?

Joe-Edwards-GDS replied:
Looks like this is an entry point that has moved out of the 'entry events' section?

There is a bit of awkwardness here - most of the usages (but not all) are to handle failures where TICF itself fails (so we don't want to call TICF _again_ before displaying the failure screen. Perhaps there's a cleverer way of doing that :shrug:

(FWIW we're looking at making the TICF call async in some way, which might help make this go away as well)

blakeyp replied:
@AmritSidhu let's keep as is for now but could you move it up to the "Entry points" section?

blakeyp commented on lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/reuse-existing-identity.yaml:
```diff
@@ -174,16 +108,6 @@ states:
-        targetState: RESET_SESSION_IDENTITY
-      end:
-        targetState: IDENTITY_REUSE_PAGE
-
```
Conveniently this factors out some of the old user-initiated reset VCs feature 🎉 ([PYIC-6394](https://govukverify.atlassian.net/browse/PYIC-6394))

AmritSidhu committed:
PYIC-8046: Reinstantiate removed test

This test is still relevant, just need to remove the IPV_GPG45_PROFILE_MATCHED audit assertions

AmritSidhu committed:
PYIC-8046: Move event up to entry point section
